optimum/habana/transformers/models/mistral/modeling_mistral.py

```diff
@@ -446,12 +521,16 @@ def forward(
         past_key_values_length = 0
         use_legacy_cache = True
         use_new_cache = False
-        if past_key_values is not None and use_cache and not reuse_cache:
-            if use_new_cache:
-                use_legacy_cache = not isinstance(past_key_values, Cache)
-                if use_legacy_cache:
-                    past_key_values = DynamicCache.from_legacy_cache(past_key_values)
-                past_key_values_length = past_key_values.get_usable_length(seq_length)
+        if past_key_values is not None and use_cache:
+            if reuse_cache:
+                # past_seen_tokens = past_key_values[0][0][2]
+                pass
```

Comment:
Why do we pass here?

Reply:
because pass_seen_tokens is not used. I adjusted the logic so it is a little less complicated now.